### PR TITLE
Move filtering, geosearch, sorting guides from `/reference/features` to `/learn/advanced`

### PR DESCRIPTION
**Pull request by guimachiavelli**
As part of our current efforts to simplify the reference page, I am slowly finding new places for content under `reference/features`.

This PR moves the guides for sorting, filtering, and geosearch to `/learn/advanced` without any changes to these pages. It also sets up the required redirects and fixes broken links caused by the move.

### CODE REVIEW

File: reference/features/README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Understanding our reference documentation is easier with background knowledge of
 [Distinct attribute](/reference/features/distinct.md)
 [Dumps](/reference/features/dumps.md)
 [Field properties](/reference/features/field_properties.md)
-[Filtering and faceted search](/reference/features/filtering_and_faceted_search.md)
+[Filtering and faceted search](/learn/advanced/filtering_and_faceted_search.md)
```
Comment (dichotommy) [inline reference/features/README.md:14]: Could we remove this link instead?

```suggestion
```

Conveniently, Geosearch and Sorting were never even added to this page! 😂 🤦🏻

**Review by dichotommy — APPROVED** [review]
LGTM 🚀

**Comment by dichotommy** [comment]
(Note that we will not merge this until the [v0.25 PR](http://github.com/meilisearch/documentation/pull/1288) has been merged, to avoid messy conflicts)

**Review by dichotommy — APPROVED** [review]
I always have time to approve a restructuring PR  🏗️

**Comment by guimachiavelli** [comment]
bors merge

**Pull request merged**